src/main/resources/META-INF/rewrite/spring-boot-40-modular-starters.yml

```diff
@@ -0,0 +1,142 @@
+#
+# Copyright 2025 the original author or authors.
+# <p>
+# Licensed under the Moderne Source Available License (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://docs.moderne.io/licensing/moderne-source-available-license
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.spring.boot4.MigrateToModularStarters
+displayName: Migrate to Spring Boot 4.0 modular starters
+description: >-
+  Adds the necessary Spring Boot 4.0 starter dependencies based on package usage. Spring Boot 4.0 has a modular design requiring explicit starters for each feature. This recipe detects feature usage via package imports and adds the appropriate starters. Note: Higher-level starters (like data-jpa) include lower-level ones (like jdbc) transitively, so only the highest-level detected starter is added for each technology.
+recipeList:
+  # spring-boot-starter-data-jpa
+  - org.openrewrite.java.ChangePackage:
+      oldPackageName: org.springframework.boot.autoconfigure.orm.jpa
+      newPackageName: org.springframework.boot.hibernate.autoconfigure
+      recursive: true
```

Comment:
Going over these now, isn't there a dependency that needs to change or be added to go along with this change? Or does this assume the still-monolithic-classic starters are present? I want to make sure folks aren't surprised by partial changes.